testplugins/src/main/java/org/spongepowered/test/MultiWorldTest.java

```diff
@@ -40,22 +40,38 @@
 
     @Override
     public void disable(CommandSource src) {
-        Sponge.getServer().getWorld("temp").ifPresent(world -> Sponge.getServer().unloadWorld(world));
+        Sponge.getServer().getWorld("no-save").ifPresent(world -> {
+            Sponge.getServer().unloadWorld(world);
+            Sponge.getServer().deleteWorld(world.getProperties());
+        });
+        Sponge.getServer().getWorld("metadata-only").ifPresent(world -> {
+            Sponge.getServer().unloadWorld(world);
+            Sponge.getServer().deleteWorld(world.getProperties());
+        });
     }
 
     @Override
     public void enable(CommandSource src) {
         try {
-            final WorldArchetype archetype = Sponge.getRegistry().getType(WorldArchetype.class, "multi-world-test:overnether").orElse(
+            final WorldArchetype archetype1 = Sponge.getRegistry().getType(WorldArchetype.class, "multi-world-test:overnether").orElseGet(() ->
                 WorldArchetype.builder().
                     from(WorldArchetypes.THE_NETHER)
                     .serializationBehavior(SerializationBehaviors.NONE)
                     .generator(GeneratorTypes.OVERWORLD)
                     .build("multi-world-test:overnether", "Overnether")
             );
-            final WorldProperties worldProperties = Sponge.getServer().createWorldProperties("temp", archetype);
+            final WorldProperties world1 = Sponge.getServer().createWorldProperties("no-save", archetype1);
+            Sponge.getServer().loadWorld(world1);
 
-            Sponge.getServer().loadWorld(worldProperties);
+            final WorldArchetype archetype2 = Sponge.getRegistry().getType(WorldArchetype.class, "multi-world-test:overend").orElseGet(() ->
```

Comment:
Should be explicit with the types here for variable names, if there's an "original" and "re-generated" or "nonSerializedArchetype" make it so, otherwise this looks like obfuscated smurf naming..